File: VB/App7/MainViewModel.vb
Imports DevExpress.Mvvm

Public Class MainViewModel
    Protected _Tasks As List(Of Task)
    Public ReadOnly Property Tasks() As List(Of Task)
        Get
            If Me._Tasks Is Nothing Then
                Me._Tasks = New List(Of Task)()
                Dim r As New Random()
                Dim i As Integer = 0
                While i < 10
                    Me._Tasks.Add(New Task() With {
                        .TaskName = "Task #" + i.ToString(),
                        .StartDate = DateTime.Today.AddDays(r.[Next](4)),
                        .EndDate = DateTime.Today.AddDays(r.[Next](10) + 4)
                    })
                    i = i + 1
                End While
            End If
            Return _Tasks
        End Get
    End Property
End Class

Public Class Task
    Inherits BindableBase

    Protected _TaskName As String
    Public Property TaskName() As String
        Get
            Return Me._TaskName
        End Get
        Set
            Me.SetProperty(Me._TaskName, Value, "TaskName")
        End Set
    End Property


    Protected _StartDate As DateTime
    Public Property StartDate() As DateTime
        Get
            Return Me._StartDate
        End Get
        Set
            Me.SetProperty(Me._StartDate, Value, "StartDate")
        End Set
    End Property


    Protected _EndDate As DateTime
    Public Property EndDate() As DateTime
        Get
            Return Me._EndDate
        End Get
        Set
            Me.SetProperty(Me._EndDate, Value, "EndDate")
        End Set
    End Property
End Class

File: CS/App6/MainViewModel.cs
using DevExpress.Mvvm;
using System;
using System.Collections.Generic;

namespace App6
{
    public class MainViewModel
    {
        protected List<Task> _Tasks;

        public List<Task> Tasks
        {
            get
            {
                if (this._Tasks == null)
                {
                    this._Tasks = new List<Task>();
                    Random r = new Random();
                    int i = 0;
                    while (++i < 10)
                    {
                        this._Tasks.Add(new Task() {
                            TaskName = "Task #" + i,
                            StartDate = DateTime.Today.AddDays(r.Next(4)), 
                            EndDate = DateTime.Today.AddDays(r.Next(10) + 4)
                        });
                    }
                }
                return _Tasks;
            }
        }
       
    }
    public class Task : BindableBase
    {

        protected string _TaskName;
        public string TaskName
        {
            get { return this._TaskName; }
            set { this.SetProperty(ref this._TaskName, value, "TaskName"); }
        }


        protected DateTime _StartDate;
        public DateTime StartDate
        {
            get { return this._StartDate; }
            set { this.SetProperty(ref this._StartDate, value, "StartDate"); }
        }


        protected DateTime _EndDate;
        public DateTime EndDate
        {
            get { return this._EndDate; }
            set { this.SetProperty(ref this._EndDate, value, "EndDate"); }
        }
    }

}
